examples/pyplots/sigmoid_function.py

```diff
@@ -0,0 +1,33 @@
+"""
+============
+Sigmoid Function
+============
+A script that showes the sigmoid function with horizontal lines for emphasis of important values.
+"""
+import numpy as np
+import matplotlib.pyplot as plt
+t = np.linspace(-10, 10, 100)
```

Comment:
I propose not to use property shortcuts like `c` or `ls` in examples to keep them easier to understand. Also to keep examples simple we're trying not to use unneccessary functions/parameters (figsize, loc do not add to the example). OTOH, I would clip the xlim to the data range because the sigmoid function is not bounded and it looks a bit weird that we have it plotted on a finite x-range and added a margin around that.

Overall, I'd go with something like this:

~~~
import numpy as np
import matplotlib.pyplot as plt

t = np.linspace(-10, 10, 100)
sig = 1 / (1 + np.exp(-t))

plt.axhline(y=0, color="black", linestyle="--")
plt.axhline(y=0.5, color="black", linestyle=":")
plt.axhline(y=1.0, color="black", linestyle="--")
plt.axvline(color="grey")
plt.plot(t, sig, linewidth=2, label=r"$\sigma(t) = \frac{1}{1 + e^{-t}}$")
plt.xlim(-10, 10)
plt.xlabel("t")
plt.legend(fontsize=14)
plt.show()
~~~

![image](https://user-images.githubusercontent.com/2836374/70862732-4f7a3b00-1f40-11ea-93a5-0de76d31a662.png)

Reply:
That looks really nice!
Thanks for this idea.